reactive/apache_bigtop_base.py

```diff
@@ -1,11 +1,24 @@
-from charms.reactive import when, when_not, set_state
-from charmhelpers.core import hookenv
+from path import Path
+from charms.reactive import when, when_any, when_not, when_none
+from charms.reactive import RelationBase, set_state
+from charms.reactive.helpers import data_changed, any_states
+from charmhelpers.core import hookenv, unitdata
 from charmhelpers.core.host import ChecksumError
 from charmhelpers.fetch import UnhandledSource
 from charms.layer.apache_bigtop_base import Bigtop
 
 
 @when('puppet.available')
+@when_none('java.ready', 'hadoop-plugin.java.ready')
+def missing_java():
+    if any_states('java.joined', 'hadoop-plugin.joined'):
+        hookenv.status_set('waiting', 'Waiting on Java')
+    else:
+        hookenv.status_set('blocked', 'Waiting on relation to Java')
+
+
+@when('puppet.available')
+@when_any('java.ready', 'hadoop-plugin.java.ready')
 @when_not('bigtop.available')
 def fetch_bigtop():
```

Comment:
`@when_any` never passes a parameter, even for states that otherwise would (because that state may or may not be set, so it just never passes them to be consistent), so I don't know how you possibly got that.